tests/tracer/test_logger.py

```diff
@@ -146,7 +145,7 @@ def test_logger_init(self):
         self.assertEqual(log.level, logging.DEBUG)
 
     def test_logger_deprecated_rate_limit(self):
-        with self.override_env(dict(DD_LOGGING_RATE_LIMIT="10")), warns(RemovedInDDTrace10Warning):
+        with self.override_env(dict(DD_LOGGING_RATE_LIMIT="10")), warns(DeprecationWarning):
```

Comment:
Debtcollector allows you to set a warning category. I can pass our custom 1.0 warning (`RemovedInDDTrace10Warning`) to a `@debtcollector.deprecate()`, `@debtcollector.removals.remove()`  and `@debtcollector.removals.removed_property()`. As of this change we are no longer using this warning category. Would it make sense to add this category to all existing warnings or just remove it from our public api?

ex:
```
@removed_module(
    module="ddtrace.compat",
    removal_version="1.0.0",
    category=ddtrace.internal.utils.deprecation.RemovedInDDTrace10Warning,
)

OR 

deprecate(
                    "DD_LOGGING_RATE_LIMIT is deprecated",
                    message="Use `DD_TRACE_LOGGING_RATE` instead",
                    removal_version="1.0.0",
                    category= ddtrace.internal.utils.deprecation.RemovedInDDTrace10Warning
                )
```

Reply:
I am fine with a generic deprecation warning?

the reason to have a specific 1.0 warning is for migration purposes... ?

Reply:
I think I am ok with this. even if someone has 1.0 and 2.0 deprecations they should just update if they can to avoid any future issues.

Reply:
I would drop the custom warning. Ideally we want to be able to remove the `ddtrace.internal.utils.deprecation` module entirely.